src/VectorEncoder.ts

```diff
@@ -187,9 +187,17 @@ export default class VectorEncoder {
       // handling more cases than we actually do
       let geometry: any = style.getGeometry();
       let geojsonFeature;
-      if (geometry) {
+      // In some cases, the geometries are objects, in other cases they're features.
+      // We need to make sure that either the geometry is an object, or that the feature it contains returns
+      // a non-null / non-undefined value.
+      if (geometry && ((geometry instanceof Object && typeof geometry === 'object') || geometry(feature))) {
```

Comment:
What about?
```suggestion
      if (geometry && ((typeof geometry === 'object') || typeof geometry === 'function')) {
```

Reply:
That would optimize a bit this check, but wouldn't ensure that the geometry the function returns is defined, which would cause issues a bit further. I would however mix them together like so : 
```if (geometry && ((typeof geometry === 'object' && geometry instanceof Object) || (typeof geometry === 'function' && geometry(feature)))  ```

this would ensure that we ensure both that it is a function and that its result is defined.